backend/src/v2v3/dex/data.ts

```diff
@@ -0,0 +1,182 @@
+import { bqClient } from "../../bigQuery";
+import { redisClient, getOrThrow } from "../../redisSetup";
+import { getErrorMessage, getPrice } from "./utils";
+
+const bigQueryEndpointBase = "crypto-stellar.crypto_stellar_2";
+
+async function fetchBigQueryData(query: string) {
+  try {
+    const [job] = await bqClient.createQueryJob(query);
+    console.log("running bq query:", query);
+    const [result] = await job.getQueryResults();
+    return result;
+  } catch (error: unknown) {
+    const message = getErrorMessage(error);
+    console.error("QUERY ERROR: ", message);
+    return null;
+  }
+}
+
+async function fetchMemoizedData(key: string, query: string) {
```

Comment:
let's just call this cached? even though memoized is technically also correct we're using `cached` throughout the codebase
```suggestion
async function fetchCached(key: string, query: string) {
```